R/dominantTaxa.R

```diff
@@ -100,9 +106,56 @@ setMethod("perSampleDominantFeatures", signature = c(x = "SummarizedExperiment")
         idx <- as.list(apply(t(mat) == colMaxs(mat),1L,which))
         # Get rownames based on indices
         taxa <- rownames(mat)[unlist(idx)]
+        
         # If multiple dominant taxa were found, names contain taxa in addition to 
         # sample name. Names are converted so that they include only sample names.
         names(taxa) <- rep( names(idx), times = lengths(idx) )
+        
+        # If individual sample contains multiple dominant taxa (they have equal counts) and if 
+        # complete is FALSE, the an arbitrarily chosen dominant taxa is returned
+        if( length(taxa) > nrow(colData(x))){
+            # Store order
+            order <- unique(names(taxa))
```

Comment:
You can get the order also from rownames(x). I'm not sure if the order matches with names(taxa) every time